lib/execution/transaction.js

```diff
@@ -219,6 +219,10 @@ class Transaction extends EventEmitter {
           return makeTransactor(this, connection, trxClient);
         })
         .then((transactor) => {
+          this.transactor = transactor;
+          if (this.outerTx) {
+            transactor.parentTransaction = this.outerTx.transactor;
```

Comment:
Please define this property in the constructor (with default undefined value) to take advantage of v8 hidden classes

Reply:
This seems unprecedented? For example `context`, `isTransaction`, `userParams`, `savepoint`, `commit`, `rollback`, `isCompleted`, and `executionPromise` are all not defined on the `QueryBuilder` constructor.